main.go

```diff
@@ -53,8 +55,23 @@ func main() {
 		}
 	}()
 
+	// Preparing RequiredAcks option, as client can't be altered after initialized
+	kgoOpts := []kgo.Opt{}
+	if cfg.Minion.EndToEnd.Enabled {
+		ack := kgo.AllISRAcks()
+		switch cfg.Minion.EndToEnd.Producer.RequiredAcks {
+		case 0:
+			ack = kgo.NoAck()
+		case 1:
+			ack = kgo.LeaderAck()
+		}
+		kgoOpts = append(kgoOpts, kgo.RequiredAcks(ack))
+		if cfg.Minion.EndToEnd.Producer.RequiredAcks != -1 {
+			kgoOpts = append(kgoOpts, kgo.DisableIdempotentWrite())
+		}
```

Comment:
RequiredAcks = -1 is all or all ISR should acknowledge, it is the default value. If the requiredAck is zero or only 1 (Only Leader Ack needed), the client init will fail since it requires idempotent by default. I think it is one of the safety feature from kgo client implementation, maybe @twmb can chip in here.

Reply:
Yes this is a safety measure. I realized while working on https://github.com/twmb/franz-go/commit/ebc8ee29528f876a0e5cea8afff050fa22d411e8 that any configuration that allows for quitting retries while producing can, in edge cases, lead to broken sequence numbers and basically a broken client.

No acks means we cannot notice when we need to retry. We may be able to allow only leader ack, but this punts the sequence number problems to edge cases in the kafka cluster.

Reply:
I see, so I guess it would be better to remove the configurable option altogether, like leave it as Idempotent and required all ISR ack? I was just wondering if there is any use case of Leader only ack or if the client will be at all usable if we allow it (Leader only ack) since from my understanding it will protrude a problem in a long run

Reply:
I'm not sure of the use case for leader only.

Unrelated, but I am in the process of updating the client such that unlimited retries are not required -- there are some scenarios where the client can know that aborting some records will not break sequence numbres.

Reply:
Noted, thanks for the info